extra-files/split_logs.py

```diff
@@ -0,0 +1,101 @@
+# -*- encoding: utf-8 -*-
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+from xml.dom import minidom
+import datetime
+import argparse
+import os
+import re
+
+def get_logs(path):
+    logs = {} # filename -> lines[]
+    for file_name in os.listdir(path):
+        if not file_name.startswith("guest-test-"):
+            continue
+        with open(path + file_name, 'r') as f:
+            logs[file_name] = f.readlines()
+    return logs
+
+def filter_logs_by_time(logs, start_time, end_time):
+    filtered_logs = {} # filename -> lines[]
+    timestamp_re = r'[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}'
+
+    def line_in_scope(line):
+        timestamp_match = re.search(timestamp_re, line)
+        if timestamp_match is None:
+            return False
+        else:
+            timestamp = datetime.datetime.fromisoformat(
+                timestamp_match.group(0))
```

Comment:
Console logs (and other system logs in general) have local timestamps, while the `start_time`/`end_time` are in UTC. You need to convert it here with `.astimezone(datetime.timezone.utc)`.

Reply:
Good point, however first I need to know the timezone the tests VM was on, correct? I was trying to find a way to know that just form the logs or openQA. Does openQA make that possible or do you know of some other way?

Reply:
The `astimezone` method converts the time from _local_ system time to selected one (here UTC), so it should work. When tested manually on one test it gives correct results.

Reply:
addressed in https://github.com/QubesOS/openqa-tests-qubesos/pull/17/commits/46f1fcc51d7cfe580c1bf50cfd3f816a1df85280